presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -822,9 +827,11 @@ public void dropColumn(HiveIdentity identity, String databaseName, String tableN
 
     private static List<String> buildPartitionNames(List<Column> partitionColumns, List<Partition> partitions)
     {
-        return partitions.stream()
-                .map(partition -> makePartitionName(partitionColumns, partition.getValues()))
-                .collect(toList());
```

Comment:
We could keep this as `toImmutableList()`. Moving away from streams simply to pre-size the output list, in code that's already doing expensive RPC calls, is a hard sell from a code readability standpoint.

Reply:
This one's a little tricky. It's not exactly the same "memoization" strategy as the rest of the PR, but it's attempting to reduce heap pressure when dealing with a large number of partitions just the same. Since `buildPartitionNames` is called from `getPartitionNamesByFilter` which might be requesting all partitions for table, the intermediate copies are material when the partition count is high.

Assuming a JVM without compressed OOPs enabled:
- 8 bytes per element in the list
- initial capacity of 4 (for `ImmutableList.builder()`, 10 for `ArrayList` although `Collectors.toList()` won't trim the resulting list to size which can be a source of wasted heap)

At 1M partitions, that's 7.6MiB for the resulting list and:
- 18 doubling copies meaning ~16MiB of allocations and copies as the builder grows
- One final 7.6MiB copy since the result is not an exact power of 2

So ~7.6MiB instead of ~23.6MiB of allocation and copying by presizing the builder, a material improvement for only a minor stylistic difference.

Reply:
That's a compelling argument -- let's keep it. One alternative idea would be to use
```java
.collect(toCollection(() -> new ArrayList(partitions.size()))
```